### DISCUSSION TITLE
Node classification on WebKB datasets

### DISCUSSION THREAD

**Discussion by AdarshMJ** [pyg-team/pytorch_geometric#5374] [Q&A] [answered]
Hi! @rusty1s I am trying to run node classification on WebKB datasets - Texas, Wisconsin and Cornell (https://pytorch-geometric.readthedocs.io/en/latest/modules/datasets.html#torch_geometric.datasets.WebKB), but I keep getting this error. 
Any idea how to resolve this?

![Screenshot 2022-09-07 at 20 02 11](https://user-images.githubusercontent.com/24205200/188947453-0e71c9a6-1adb-4fca-9396-3e9c122e02af.png)

Thank you in advance!

**Comment by EdisonLeeeee** [comment]
Hi @AdarshMJ 

There are ten different splits in `WebKB` so the `train_mask` (or `test_mask`) got the shape `[183, 10]`. You need to specify the `split_idx` for a given `mask`:
```python
# `split_idx` could be 0, 1, ..., 9
mask = data.train_mask[:, split_idx]
loss =  criterion(out[mask], data.y[mask])
```

**Reply by AdarshMJ** [reply]
Hi @EdisonLeeeee Thank you for your response. I have got it to working, but the accuracy is very low, heres the code I am working with -

The model is a simple GCN.
```Python
GCN(
  (conv1): GCNConv(1703, 32)
  (conv2): GCNConv(32, 5)
)
```
```Python

def train():
      model.train()
      optimizer.zero_grad()  # Clear gradients.
      out = model(data.x, data.edge_index)  # Perform a single forward pass.
      loss = criterion(out[data.train_mask[:, 0]], data.y[data.train_mask[:, 0]])  # Compute the loss solely based on the training nodes.
      loss.backward()  # Derive gradients.
      optimizer.step()  # Update parameters based on gradients.
      return loss

def test():
      model.eval()
      out = model(data.x, data.edge_index)
      pred = out.argmax(dim=1)  # Use the class with highest probability.
      correct = pred[data.val_mask[:, 0]] == data.y[data.val_mask[:, 0]]  # Check against ground-truth labels.
      test = int(correct.sum()) / int(data.val_mask.sum())  # Derive ratio of correct predictions.
      return test


for epoch in range(1, 101):
    loss = train()
    #print(f'Epoch: {epoch:03d}, Loss: {loss:.4f}')
test = test()
print(f'Test Accuracy: {test:.4f}')
```

The accuracy I get is very low - `Test Accuracy: 0.0610`

Do you think I should play around with the hyperparameters or is there something wrong with my code? Any insights would be helpful! Thank you.

**Comment by EdisonLeeeee** [comment, marked as answer]
In `test`, the computation of accuracy should be
```python
test = int(correct.sum()) / int(data.val_mask[:, 0].sum())
```
and you'll get your accuracy of 0.61

**Reply by AdarshMJ** [reply]
Thank you!
